src/Arcus.Templates.AzureFunctions.EventHubs/SensorReadingFunction.cs

```diff
@@ -20,60 +23,99 @@ namespace Arcus.Templates.AzureFunctions.EventHubs
-
-        public SensorReadingFunction(IAzureEventHubsMessageRouter messageRouter)
+#if InProcess
+        private readonly AzureFunctionsInProcessMessageCorrelation _messageCorrelation;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SensorReadingFunction"/> class.
+        /// </summary>
+        /// <param name="messageRouter">The message router instance to route the Azure EventHubs events through the sensor-reading processing.</param
+        /// <param name="messageCorrelation">The message correlation instance to W3C correlate the Azure EventHubs events.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="messageRouter"/> or the <paramref name="messageCorrelation"/> is <c>null</c>.</exception>
+        public SensorReadingFunction(
+            IAzureEventHubsMessageRouter messageRouter,
+            AzureFunctionsInProcessMessageCorrelation messageCorrelation)
         {
+            Guard.NotNull(messageRouter, nameof(messageRouter), "Requires a message router instance to route incoming Azure EventHubs events through the sensor-reading processing");
+            Guard.NotNull(messageCorrelation, nameof(messageCorrelation), "Requires a message correlation instance to W3C correlate incoming Azure EventHubs events");
+            
             _messageRouter = messageRouter;
+            _messageCorrelation = messageCorrelation;
         }
-
-#if InProcess
+        
+        /// <summary>
+        /// Processes Azure EventHubs <paramref name="events"/>.
+        /// </summary>
+        /// <param name="events">The incoming events on the Azure EventHubs instance.</param>
+        /// <param name="log">The logger instance to write informational messages during the message processing.</param>
+        /// <param name="cancellationToken">The token to cancel the message processing.</param>
         [FunctionName("sensor-reading")]
         public async Task Run(
             [EventHubTrigger("sensors", Connection = "EventHubsConnectionString")] EventData[] events,
             ILogger log,
             CancellationToken cancellationToken)
         {
+            log.LogInformation("Azure EventHubs function triggered with {Length} events", events.Length);
             foreach (EventData message in events)
             {
-                log.LogInformation($"First Event Hubs triggered message: {message.MessageId}");
-
-                var messageContext = AzureEventHubsMessageContext.CreateFrom(message, "sensor-reading.servicebus.windows.net", "$Default", "sensors");
-                MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
-                await _messageRouter.RouteMessageAsync(message, messageContext, correlationInfo, cancellationToken); 
+                AzureEventHubsMessageContext messageContext = message.GetMessageContext("sensor-reading.servicebus.windows.net", "sensors");
```

Comment:
Just wondering if 'servicebus' is correct here ?

Reply:
Yes, it is. This is how the Azure EventHubs connection string looks like: https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string